lib/AutomaticPrefetchPlugin.js

```diff
@@ -2,34 +2,35 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var async = require("async");
-var PrefetchDependency = require("./dependencies/PrefetchDependency");
-var NormalModule = require("./NormalModule");
+"use strict";
 
-function AutomaticPrefetchPlugin() {}
-module.exports = AutomaticPrefetchPlugin;
-AutomaticPrefetchPlugin.prototype.apply = function(compiler) {
-	compiler.plugin("compilation", function(compilation, params) {
-		var normalModuleFactory = params.normalModuleFactory;
+const async = require("async");
+const PrefetchDependency = require("./dependencies/PrefetchDependency");
+const NormalModule = require("./NormalModule");
+
+class AutomaticPrefetchPlugin {
+	apply(compiler) {
+		compiler.plugin("compilation", (compilation, params) => {
+			const normalModuleFactory = params.normalModuleFactory;
 
-		compilation.dependencyFactories.set(PrefetchDependency, normalModuleFactory);
-	});
-	var lastModules = null;
-	compiler.plugin("after-compile", function(compilation, callback) {
-		lastModules = compilation.modules.filter(function(m) {
-			return m instanceof NormalModule;
-		}).map(function(m) {
-			return {
-				context: m.context,
-				request: m.request
-			};
+			compilation.dependencyFactories.set(PrefetchDependency, normalModuleFactory);
+		});
+		let lastModules = null;
+		compiler.plugin("after-compile", (compilation, callback) => {
```

Comment:
Supporting both has probably negative effect on performance, but I think it's possible to update everything to Promises (in future)...